src/Interpreters/GinFilter.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <Storages/MergeTree/GinIndexStore.h>
+namespace DB
+{
+struct GinFilterParameters
+{
+    explicit GinFilterParameters(size_t ngrams_);
+
+    size_t ngrams;
+};
+
+struct RowIDRange
+{
+    /// Segment ID of the row ID range
+    UInt32 segment_id;
+
+    /// First row ID in the range
+    UInt32 range_start;
+
+    /// Last row ID in the range
+    UInt32 range_end;
+};
+
+class GinFilter
+{
+public:
+    using RowIDRangeContainer = std::vector<RowIDRange>;
+
+    explicit GinFilter(const GinFilterParameters& params);
+
+    void add(const char* data, size_t len, UInt32 rowID, GinIndexStorePtr& store);
```

Comment:
Please add comments to each non-trivial method of the GinFilter class.

Reply:
Fixed.